Algorithms and Data Structures/TreasureHunt/TreasureHunt/Map.cs

```diff
@@ -0,0 +1,149 @@
+                {
+                    _map[i, j] = random.Next(10);
+                }
+            }
+        }
+
+        private void FindTreasure()
+        {
+            _treasurePosition = FindMaxValueIndex();
+        }
+
+        private void FindTrap()
+        {
+            _trapPosition = FindMinValueIndex();
+        }
+
+        private int[] FindMaxValueIndex()
+        {
+            var maxValue = _map[0,0];
+            var maxValueIndex = new int[2];
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    if (_map[i, j] > maxValue)
+                    {
+                        maxValue = _map[i, j];
+                        maxValueIndex = new[] { i, j };
+                    }
+                }
+            }
+
+            _maxValue = maxValue;
+            return maxValueIndex;
+        }
+
+        private int[] FindMinValueIndex()
+        {
+            var minValue = _map[0, 0];
+            var minValueIndex = new int[2];
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    if (_map[i, j] < minValue)
+                    {
+                        minValue = _map[i, j];
+                        minValueIndex = new[] { i, j };
+                    }
+                }
+            }
+
+            _minValue = minValue;
+            return minValueIndex;
+        }
+
+        public void DisplayMap()
+        {
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    Console.Write($"[{_map[i, j]}]");
+                }
+                Console.WriteLine();
+            }
+        }
+
+        public void DisplayMapWithPlayer(Player player)
+        {
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    if (player.PlayerX == i && player.PlayerY == j)
+                    {
+                        Console.Write("[P]");
+                    }
+                    else
+                    {
+                        Console.Write("[0]");
+                    }
+                }
+                Console.WriteLine();
+            }
+        }
+
+        private void OptimizeMap()
+        {
+            for (var i = 0; i < _mapX; i++)
+            {
+                for (var j = 0; j < _mapY; j++)
+                {
+                    if (_map[i, j] == _maxValue && i != TreasureX && j != TreasureY)
+                    {
+                        _map[i, j]--;
+                    }
+                    
+                    if (_map[i, j] == _minValue && i != TrapX && j != TrapY)
+                    {
+                        _map[i, j]++;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
we do optimization when we need to, we dont need any here.

keep in mind that we call it optimization when it really optimize the calculation, this just adds some useless calculation to clean the data that adds to the load.

also these sorts of stuff will make the reader more confused.